### PR TITLE
[Owl] Migrates navigation to navigation-compose

### PR DESCRIPTION
**Pull request by JoseAlcerreca**
Also adds new navigation tests

TODO: Bottom nav migration to navigation-compose

### CODE REVIEW

File: Owl/app/src/androidTest/java/com/example/owl/ui/NavigationTest.kt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        // The first course should be shown
+        composeTestRule.onNodeWithSubstring(courses.first().name).assertExists()
+    }
+
+    @Test
+    fun coursesToDetail() {
+        // Given the app in the courses screen
+        startActivity(MainDestinations.COURSES_ROUTE)
+
+        composeTestRule.onRoot().printToLog("jalc")
```
Comment (nickbutcher) [inline Owl/app/src/androidTest/java/com/example/owl/ui/NavigationTest.kt:93]: Remove

File: Owl/app/src/androidTest/java/com/example/owl/ui/NavigationTest.kt [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import com.example.owl.ui.fakes.ProvideTestImageLoader
+import com.example.owl.ui.utils.AmbientBackDispatcher
+import dev.chrisbanes.accompanist.insets.ProvideWindowInsets
+import org.junit.Rule
+import org.junit.Test
+
+/**
+ * Checks that the navigation flows in the app are correct.
+ */
+class NavigationTest {
```
Comment (nickbutcher) [inline Owl/app/src/androidTest/java/com/example/owl/ui/NavigationTest.kt:41]: 👏👏👏

File: Owl/app/src/androidTest/java/com/example/owl/ui/fakes/ProvideTestImageLoader.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import coil.memory.MemoryCache
+import coil.request.DefaultRequestOptions
+import coil.request.Disposable
+import coil.request.ImageRequest
+import coil.request.ImageResult
+import coil.request.SuccessResult
+import dev.chrisbanes.accompanist.coil.AmbientImageLoader
+
+/**
+ * Replaces all images with a simple black drawable to make testing faster and hermetic.
```
Comment (nickbutcher) [inline Owl/app/src/androidTest/java/com/example/owl/ui/fakes/ProvideTestImageLoader.kt:36]:
```suggestion
 * Replaces all remote images with a simple black drawable to make testing faster and hermetic.
```

File: Owl/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -32,6 +32,7 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name="androidx.activity.ComponentActivity" />
```
Comment (nickbutcher) [inline Owl/app/src/main/AndroidManifest.xml:35]: Is this for tests? Does this have to be in the main src set?

File: Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt [hunk 5]
```diff
@@ -16,37 +16,70 @@
...
+import androidx.navigation.compose.navigate
+import androidx.navigation.compose.rememberNavController
+import com.example.owl.ui.MainDestinations.COURSE_DETAIL_ID_KEY
+import com.example.owl.ui.course.CourseDetails
+import com.example.owl.ui.courses.Courses
+import com.example.owl.ui.onboarding.Onboarding
 
 /**
- * Models the screens in the app and any arguments they require.
+ * Destinations used in the main screen ([OwlApp]).
```
Comment (nickbutcher) [inline Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt:34]: This seems misleading – Onboarding isn't on the main screen

File: Owl/app/src/androidTest/java/com/example/owl/ui/fakes/ProvideTestImageLoader.kt [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+ */
+@OptIn(ExperimentalCoilApi::class)
+@Composable
+fun ProvideTestImageLoader(content: @Composable () -> Unit) {
```
Comment (nickbutcher) [inline Owl/app/src/androidTest/java/com/example/owl/ui/fakes/ProvideTestImageLoader.kt:40]: This is neat!

File: Owl/app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (JoseAlcerreca) [reply Owl/app/src/main/AndroidManifest.xml:35]: Ah forgot to move

File: Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt (hunk 5 above)
Comment (JoseAlcerreca) [reply Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt:34]: Fixed but let's revisit naming when we include the bottom bar, I'm not sure if it's a subgraph or we'll just add destinations here.

**Review by nickbutcher — APPROVED** [review]

**Pull request merged**